test/controllers/actions/MutableFakeDataRetrievalAction.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.actions
+
+import models.UserAnswers
+import models.requests.{IdentifierRequest, OptionalDataRequest}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class MutableFakeDataRetrievalAction extends DataRetrievalAction {
+  def setDataToReturn(userAnswers: Option[UserAnswers]): Unit = dataToReturn = userAnswers
+
+  private var dataToReturn: Option[UserAnswers] = None
+
+  override protected def transform[A](request: IdentifierRequest[A]): Future[OptionalDataRequest[A]] =
+    dataToReturn match {
```

Comment:
Don't need to bother doing a match here. Just return optional data request with whatever is in dataToReturn.

Reply:
Done